.github/workflows/pr-build-image.yaml

```diff
@@ -27,8 +13,27 @@ env:
   REGISTRY: quay.io
 
 jobs:
+  check-image:
+    name: Check Image and Relevant Changes
+    runs-on: ubuntu-latest
+    outputs:
+      image_exists: ${{ steps.check-image.outputs.image_exists }}
+      relevant_changes: ${{ steps.check-image.outputs.relevant_changes }}
+      short_sha: ${{ steps.check-image.outputs.short_sha }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check Image and Relevant Changes
+        id: check-image
+        uses: ./.github/actions/check-image-and-changes
+
   build-image:
     name: Build Image
+    needs: check-image
+    if: ${{ needs.check-image.outputs.relevant_changes == 'true' || needs.check-image.outputs.image_exists == 'false' }}
```

Comment:
this is a required test, so the same "must run and if skipped should exit 0" rule applies here too. 

![image](https://github.com/user-attachments/assets/0caed329-39b9-4e07-8ec9-01a502c92988)